### ISSUE TITLE
Perspective node runtime

### ISSUE THREAD

**Issue by LukeSheard** [jpmorganchase/perspective#243]
When trying to use the node build of perspective the javascript errors out on initialization:

```javascript
const perspective = = require("@jpmorganchase/perspective").default
Object.keys(perspective)
/* 
[ 'worker',
  'TYPE_AGGREGATES',
  'TYPE_FILTERS',
  'AGGREGATE_DEFAULTS',
  'FILTER_DEFAULTS',
  'SORT_ORDERS' ]
*/
const worker = perspective.worker()
/*
ReferenceError: window is not defined
    at Object.worker (/Users/f608687/dev/scratch/node_modules/@jpmorganchase/perspective/build/perspective.umd.js:1:8192)
*/
```

Would it be best to remove the webpack build for node and maybe spend time later on writing a proper node-gyp binding the perspective C++ runtime ala #212 ?

**Comment by texodus** [comment]
Thanks for the report!

This certainly needs better documentation and some code cleanup.  The node.js runtime does not support a worker implementation currently, due to

*  The API differences between WebWorkers and Node's `child_process` module made this not free (unless there is a better way).  
* There is no renderer process in node.js, so I was originally unclear that subprocess instantiation of the node.js was even warranted.

The intention of the documentation was to show the 2 separate ways to access the the `perpsective` module symbol, [In The Browser](https://jpmorganchase.github.io/perspective/docs/usage.html#in-the-browser) for using the `worker()` method to get a `perspective` module symbol, and [In Node.js](https://jpmorganchase.github.io/perspective/docs/usage.html#in-nodejs) for acquiring the same symbol simply by importing it.  This is wildly insufficient, so I'll be cleaning up the documentation to make this easier to understand and will close this issue once it is merged.

Separately, I am completely open to adding a subprocess API to the node.js runtime as well.  If you're interested in contributing such a change, PRs are always welcome - if you would just like to request and track progress on this, feel free to open a separate issue detailing your use case so we can better understand how best to implement it.

**Comment by texodus** [comment]
Regarding `node-gyp` - I think we'd need some clear explanation of what this would bring to the project over the WebAssembly runtime.  From my (albeit limited) scope, `node-gyp` would seem to add quite alot of additional build complexity, potentially a separate c++ API for binding, and a loss of portability;  while the performance gain has at best not been explored, and I have reason to suspect it is minimal in Perspective's case.
